geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+    kind: Gateway
+    group: gateway.networking.k8s.io
+  listeners:
+  - name: second
+    hostname: second.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: second-workload-cert # Provisioned via HTTP01 challenge
+```
+### Semantics
+
+#### Gateway <> ListenerSet Handshake
+
+By default a `Gateway` will allow `ListenerSets` in the same namespace to be attached. Users can prevent this behaviour by configuring their `Gateway` to disallow any listener attachment:
+
+```
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  allowedListeners:
+  - from: None
+```
+
+#### Listeners
+
+####  `ListenerSet.spec.parentRef` is immutable
+
+A `ListenerSet` `MUST` be created with a `parentRef` Updating the `parentRef` of a ListenerSet `MUST` be rejected by the API.
+
+#### `ListenerSet` Default Labels
+
+In order for clients of the API to easily discover `Gateways` and their `ListenerSets` we propose a set of labels to be defaulted when creating `ListenerSets`. This only applies if the `parentRef.kind` is a `Gateway`
+
+- `gateway.networking.k8s.io/gateway.name` is equal to `spec.parentRef.name`
+- `gateway.networking.k8s.io/gateway.namespace` is equal to `spec.parentRef.namespace`
+
+This allows clients to perform API [lookups](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#list-and-watch-filtering) using label selectors.
+
+TBD: Can conditional defaulting be done with CEL?
+
+#### GatewaySpec.Listeners MinItems
+
+TBD: Do we want to make this change?
+
+Currently when creating a Gateway you must specify at least a single listener. With `ListenerSets` this opens the possibility of wanting to create a Gateway with no listeners.
+
+#### Route Attaching
+
+Routes MUST be able to specify a `ListenerSet` as a `parentRef` and make use of the `sectionName` field in `ParentReference` to help target a specific listener. If no listener is targeted (`sectionName`/`port` are unset) then the Route references all the listeners on the `ListenerSet`. It `MUST NOT` attach to additional listeners on the parent `Gateway`.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: second-workload-listeners
+    kind: ListenerSet
+    sectionName: second
+```
+
+For instance, the following `HTTPRoute` attemps to attach to a listener defined in the parent `Gateway`. This is not valid and the route's status should reflect that.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: some-workload-listeners
+    kind: ListenerSet
+    sectionName: foo
+```
+
+#### Listener Validation
+
+Implementations MUST treat the parent `Gateway` as having the concatenated list of all listeners from itself and attached `ListenerSets`
```

Comment:
It seems like for this to work, we'll need to make the order in which Listeners are specified meaningful (earlier in the list has precedence), and also define the order in which ListenerSets should be concatenated.

Reply:
it's defined in another section. added a comment.